sys/include/xtimer.h

```diff
@@ -0,0 +1,442 @@
+ */
+#ifndef XTIMER_H
+#define XTIMER_H
+
+#include <stdint.h>
+#include "msg.h"
+#include "periph/timer.h"
+#include "timex.h"
+
+#include "board.h"
+#include "periph_conf.h"
+
+/**
+ * @brief internal define to allow using variables instead of defines
+ */
+#ifdef XTIMER_TRACE
+#include "xtimer_trace.h"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief xtimer callback type
+ */
+typedef void (*timer_callback_t)(void*);
+
+/**
+ * @brief xtimer timer structure
+ */
+typedef struct xtimer {
+    struct xtimer *next;        /**< reference to next timer in timer lists */
+    uint32_t target;            /**< lower 32bit absolute target time */
+    uint32_t long_target;       /**< upper 32bit absolute target time */
+    timer_callback_t callback;  /**< callback function to call when timer expires */
+    void *arg;                  /**< argument to pass to callback function */
+} xtimer_t;
+
+/**
+ * @brief get the current system time as 32bit microsecond value
+ *
+ * @return current time as 32bit microsecond value
+ */
+static inline uint32_t xtimer_now(void);
+
+/**
+ * @brief get the current system time as 64bit microsecond value
+ *
+ * @return current time as 64bit microsecond value
+ */
+uint64_t xtimer_now64(void);
+
+/**
+ * @brief get the current system time into a timex_t
+ *
+ * @param[out] out pointer to timex_t the time will be written to
+ */
+void xtimer_now_timex(timex_t *out);
+
+/**
+ * @brief xtimer initialization function
+ *
+ * This sets up xtimer. Has to be called once at system boot.
+ * If @ref auto_init is enabled, it will call this for you.
+ */
+void xtimer_init(void);
+
+/**
+ * @brief Stop execution of a thread for some time
+ *
+ * When called from an ISR, this function will spin and thus block the MCU in
+ * interrupt context for the specified amount in *seconds*, so don't *ever* use
+ * it there,
+ *
+ * @param[in] seconds the amount of seconds the thread should sleep
+ */
+static void xtimer_sleep(uint32_t seconds);
+
+/**
+ * @brief Stop execution of a thread for some time
+ *
+ * When called from an ISR, this function will spin and thus block the MCU for
+ * the specified amount in microseconds, so only use it there for *very* short
+ * periods, e.g., less than XTIMER_BACKOFF.
+ *
+ * @param[in] microseconds the amount of microseconds the thread should sleep
+ */
+static void xtimer_usleep(uint32_t microseconds);
+
+/**
+ * @brief Stop execution of a thread for some time, 64bit version
+ *
+ * When called from an ISR, this function will spin and thus block the MCU for
+ * the specified amount in microseconds, so only use it there for *very* short
+ * periods, e.g., less then XTIMER_BACKOFF.
+ *
+ * @param[in] microseconds the amount of microseconds the thread should sleep
+ */
+static inline void xtimer_usleep64(uint64_t microseconds);
+
+/**
+ * @brief Stop execution of a thread for some time
+ *
+ * Don't expect nanosecond accuracy. As of mid 2015, this function just calls
```

Comment:
Why not simply `as of now`? Whoever and whenever the implementation gonna change, this comment can also be updated.